packages/core/src/controller/Controller.ts

```diff
@@ -0,0 +1,128 @@
+import type { ActionTypes } from '@rest-hooks/core/types';
+import type { EndpointInterface, ResolveType } from '@rest-hooks/endpoint';
+import createInvalidate from '@rest-hooks/core/controller/createInvalidate';
+import createFetch from '@rest-hooks/core/controller/createFetch';
+import createReset from '@rest-hooks/core/controller/createReset';
+import createReceive from '@rest-hooks/core/controller/createReceive';
+import {
+  createUnsubscription,
+  createSubscription,
+} from '@rest-hooks/core/controller/createSubscription';
+import type { EndpointUpdateFunction } from '@rest-hooks/core/controller/types';
+import type { DenormalizeCache } from '@rest-hooks/normalizr';
+
+type RHDispatch = (value: ActionTypes) => Promise<void>;
+
+interface ConstructorProps {
+  dispatch: RHDispatch;
+  globalCache?: DenormalizeCache;
+}
+
+/**
+ * Imperative control of Rest Hooks store
+ * @see https://resthooks.io/docs/api/Controller
+ */
+export default class Controller {
+  declare readonly dispatch: RHDispatch;
+  declare readonly globalCache: DenormalizeCache;
+
+  constructor({
+    dispatch,
+    globalCache = {
+      entities: {},
+      results: {},
+    },
+  }: ConstructorProps) {
+    this.dispatch = dispatch;
+    this.globalCache = globalCache;
+  }
+
+  /*************** Action Dispatchers ***************/
+
+  fetch = <
+    E extends EndpointInterface & { update?: EndpointUpdateFunction<E> },
+  >(
+    endpoint: E,
+    ...args: readonly [...Parameters<E>]
+  ): ReturnType<E> => {
+    const action = createFetch(endpoint, {
+      args,
+    });
+    this.dispatch(action);
+
+    return action.meta.promise as any;
```

Comment:
maybe?
```suggestion
    return action.meta.promise as Promise<ReturnType<E>>;
```

Reply:
ReturnType<E> is already a promise so i'll just put this in manually